demo_nodes_cpp/CMakeLists.txt

```diff
@@ -77,51 +77,149 @@ add_dependencies(services_library)
-  PLUGIN "demo_nodes_cpp::ListParameters"
-  EXECUTABLE list_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterBlackboard"
-  EXECUTABLE parameter_blackboard)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::SetAndGetParameters"
-  EXECUTABLE set_and_get_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterEventsAsyncNode"
-  EXECUTABLE parameter_events_async)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::EvenParameterNode"
-  EXECUTABLE even_parameters_node)
-
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Talker"
-  EXECUTABLE talker)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageTalker"
-  EXECUTABLE talker_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Listener"
-  EXECUTABLE listener)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageListener"
-  EXECUTABLE listener_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::ListenerBestEffort"
-  EXECUTABLE listener_best_effort)
+rclcpp_components_register_nodes(
+  PLUGINS
+    "demo_nodes_cpp::ReuseTimerNode"
+    "demo_nodes_cpp::OneOffTimerNode"
+  FROM timers_library
+)
+
+rclcpp_components_add_executable(one_off_timer
+  PLUGINS "demo_nodes_cpp::ReuseTimerNode"
+  FROM timers_library
+)
+
+rclcpp_components_add_executable(reuse_timer
+  PLUGINS "demo_nodes_cpp::ReuseTimerNode"
+  FROM timers_library
+)
+
+install(TARGETS
+  one_off_timer reuse_timer
+  DESTINATION lib/${PROJECT_NAME}
+)
+
+rclcpp_components_register_nodes(
+  PLUGINS
+    "demo_nodes_cpp::ServerNode"
+    "demo_nodes_cpp::ClientNode"
+  FROM services_library
+)
+rclcpp_components_add_executable(add_two_ints_server
+  PLUGINS "demo_nodes_cpp::ServerNode"
+  FROM services_library
+)
+rclcpp_components_add_executable(add_two_ints_client_async
+  PLUGINS "demo_nodes_cpp::ClientNode"
+  FROM services_library
+)
+
+install(TARGETS
+  add_two_ints_server add_two_ints_client_async
+  DESTINATION lib/${PROJECT_NAME}
+)
+
+rclcpp_components_register_nodes(
+  PLUGINS
+    "demo_nodes_cpp::ListParameters"
+    "demo_nodes_cpp::ParameterBlackboard"
+    "demo_nodes_cpp::SetAndGetParameters"
+    "demo_nodes_cpp::ParameterEventsAsyncNode"
+    "demo_nodes_cpp::EvenParameterNode"
+  FROM parameters_library
+)
+
+rclcpp_components_add_executable(list_parameters
+  PLUGINS "demo_nodes_cpp::ListParameters"
+  FROM parameters_library
+)
+
+rclcpp_components_add_executable(parameter_blackboard
+  PLUGINS "demo_nodes_cpp::ParameterBlackboard"
+  FROM parameters_library
+)
+
+# rclcpp_components_add_executable(compound_server
+#   PLUGINS
+#     "demos_nodes_cpp::ServerNode"
+#   FROM services_library
+# )
+#
+# rclcpp_components_target_compose(compound_server
```

Comment:
How does this work? Does it add to `compound_server`?

Reply:
> Does it add to compound_server?

Correct. There's a small gap to bridge if we go from 

```cmake
rclcpp_components_add_executable(my_executable
   COMPONENT "some_ns::Talker"
   FROM my_library 
)
```

to

```cmake
rclcpp_components_add_executable(my_executable
   COMPONENTS 
     "some_ns::Talker"
     "some_ns::Listener"
   FROM my_library 
)
```

It gets somewhat larger if we want to enable adding components from multiple libraries. That's 
`rclcpp_components_target_compose()` purpose. Naming suggestions are much appreciated.

Reply:
I don't think that we have the ability to support the second case, correct?  The current setup only allows for the 1-to-1 mapping of component to executable?

Reply:
> The current setup only allows for the 1-to-1 mapping of component to executable?

Correct, but jumping from one component in one library to N components in M libraries is straightforward. Not something that we have to address now though, but we can gear the API in that direction and add missing bits later on.